core/src/display_object/movie_clip.rs

```diff
@@ -4063,6 +4209,59 @@ impl<'gc, 'a> MovieClipData<'gc> {
         Ok(())
     }
 
+    #[inline]
+    fn get_registered_character_by_id(
+        &mut self,
+        context: &mut UpdateContext<'_, 'gc>,
+        id: CharacterId,
+    ) -> Result<Character<'gc>, Error> {
```

Comment:
Can this return an `Option<Character>` instead, and log an error if the character isn't found?